README.md

```diff
@@ -218,6 +219,50 @@ Note that First-Party Sets also gives browsers the opportunity to group per-site
 those at `chrome://settings/content/all`) by the “first-party” boundary instead of eTLD+1, which is 
 not always the correct site boundary.
 
+# Clearing Site Data on Set Transitions
+Sites can switch the First-Party Set that they are a member of. We need to pay attention to these transitions so that they don’t link user identities across all the FPSs they’ve historically been in. The concrete goals that we aim to achieve are as follows:
+
+*   Preserve user expectations that their browsing activity is scoped/partitioned to “party” (as defined by First-Party Set).
+*   Prevent a potential privacy attack that uses a throwaway domain to rotate between sets, and therefore sync user data or identifiers across disparate sets. 
+
+In order to achieve this, site data needs to be cleared on certain transitions., The clearing should behave like [Clear-Site-Data](https://www.w3.org/TR/clear-site-data/), for which site data includes cookies, storage, cache, as well as execution contexts (documents, workers, etc.). We don’t differentiate between different types of site data since a site could expect different types of data to co-exist and require some consistency.
+
+Sites switching FPSs can be viewed as “leaving one FPS and joining another”, and the member sites can only add/remove themselves to/from FPSs with the consent from the owners. If a FPS’s owner changes, that set is considered dissolved. Thus we associate the need to clear site data with whether that site’s FPS owner has changed, with a few exceptions.
+
+If a site’s owner changed:
+
+1. If this site is joining a FPS, no site data clearing will be enforced.
+    *   Pro: Avoids adoption pain when a site first joins a FPS.
+    *   Con: Unclear how this lines up with user expectations about access to browsing history prior to set formation.
+2. If this site is leaving a FPS, clear site data of this site.
+
+Potential modification, which adds implementation complexity:
+
+3. If this site’s new owner used to be a member site from the same FPS, no site data clearing will be enforced. 
+    *   Pro: Provides graceful transitions for examples (f) and (g).
+    *   Con: Multi-stage transitions, such as (h) to (i) are unaccounted for.
+
+## Examples
+
+![Clear Site Data Doc Img_ 1  FPS representation](https://user-images.githubusercontent.com/89418275/130630675-d1b7a836-afe8-4746-bf1a-e5e216a92032.png)
```

Comment:
> Can you also check in source of some sort for these images, so that it's easy to modify them in the future? I've seen https://plantuml.com/ and https://mermaid-js.github.io/ used for this.

https://marketplace.visualstudio.com/items?itemName=hediet.vscode-drawio might also be a good option. I believe its `.drawio.svg` format is what you'd get by using the 'Editable Vector Image' save format in https://app.diagrams.net/.